### ISSUE TITLE
Edge runtime timeout with drizzle and vercel postgres

### ISSUE THREAD

**Issue by Xexr** [vercel/next.js#49918]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP Fri Jan 27 02:56:13 UTC 2023
    Binaries:
      Node: 16.18.0
      npm: 8.19.2
      Yarn: N/A
      pnpm: 8.5.1
    Relevant packages:
      next: 13.4.3-canary.1
      eslint-config-next: 13.4.1
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Middleware / Edge (API routes, runtime)

### Link to the code that reproduces this issue

https://github.com/Xexr/next_test

### To Reproduce

Set vercel postgres env vars (ignore the clerk variable)

Install packages with pnpm install

Navigate to http://localhost:3000

Refresh page 10 or so times and you will see that next hangs and the page doesn't refresh.

You can also see the behaviour at: https://next-drizzle-test.vercel.app/

### Describe the Bug

When using the edge runtime with server actions, drizzle and vercel postgres, data fetching/writing eventually causes next to hang after 5-10 times.

I can't work out what exactly is causing the hang, but when I disable the edge runtime and use the node runtime, the issue does not exist.

The issue occurs in both development and when deployed to vercel.

### Expected Behavior

I should be able to fetch data / refresh the page as many times as I like without next hanging.

### Which browser are you using? (if relevant)

Brave Version 1.51.114 Chromium: 113.0.5672.92 (Official Build) (64-bit)

### How are you deploying your application? (if relevant)

Vercel

**Comment by Xexr** [comment]
For anyone who finds this. I believe the issue is with the drizzle postgres plugin or the vercel storage driver.

I've switched to planetscale mysql, using drizzle and next edge functions and the issue doesn't repeat.

**Comment by alexis-dorosz** [comment]
I have the same issue using drizzle-orm and vercel postgres
I'm surprised this is not getting more attention

If you refresh a few times in quick succession edge function timeouts and the website is down

**Comment by Xexr** [comment]
Yes exactly, I was also surprised by the radio silence as it seems like a setup that is going to be prevalent (next edge, Vercel Postgres, drizzle). 

Also asked in the next and drizzle discord help forums and again, silence in both. Pretty disappointing. 

I’ll reopen the issue as I realise it’s not actually solved, I’ve just replaced vercel Postgres to get around it. 

I’ll try and see if I can work out if the issue is vercel Postgres or drizzle pg to help narrow things down.

**Comment by alexis-dorosz** [comment]
I tried with neondatabase directly (what vercel uses under the hood) and the same issue occurs
However just like you I have no problem when switching to planetscale

Probably not a vercel issue in this case

**Comment by chungweileong94** [comment]
I was facing the same issue with neon.tech, which is what Vercel postgres is using. And I strongly believe that it's either caused by neondatabse driver or the neon services.

**Comment by markteekens** [comment]
i'm using nuxt with planetscale, vercel and got a server timeout. Probably cold start is too long??

**Comment by Xexr** [comment]
I think this particular issue is with neon (and by extension vercel storage). 

Planetscale and drizzle using the planetscale js driver has been flawless for me so far.

**Comment by ghostneneji** [comment]
cold start + 10sec edge function timeout is lethal combination, this is asbolutely not reliable solution

**Comment by vvo** [comment]
Hi to all of you in this thread, we've identified a problematic pattern of `@vercel/postgres` that indeed generates timeout when deployed on Vercel Edge Functions.

While we dig this issue we advise you to switch **from:**

```ts
import { sql } from "@vercel/postgres";

const client = await sql.connect();
const result = await client.query('SELECT now()');
```
**👆This is currently triggering timeouts on Edge Function, do not use it if you can avoid it.**

**to:**
```ts
import { sql } from "@vercel/postgres";

const result = await sql`SELECT now()`;
// or const result = await sql.query('SELECT now()');
```

**👆This works and doesn't trigger timeouts.**

We understand that this is not ideal, especially because the drizzle-orm currently uses the problematic pattern.
We understand not everyone can switch to this pattern.

> I tried with neondatabase directly (what vercel uses under the hood) and the same issue occurs However just like you I have no problem when switching to planetscale
> 
> Probably not a vercel issue in this case

Dear @chungweileong94 and @alexis-dorosz, if you were able to reproduce this issue directly on neon.tech, could you provide us with some code samples you used that reproduced the issue? Thanks.

**Comment by ghostneneji** [comment]
> Hi to all of you in this thread, we've identified a problematic pattern of `@vercel/postgres` that indeed generates timeout when deployed on Vercel Edge Functions.
> 
> While we dig this issue we advise you to switch **from:**
> 
> ```ts
> import { sql } from "@vercel/postgres";
> 
> const client = await sql.connect();
> const result = await client.query('SELECT now()');
> ```
> 
> **👆This is currently triggering timeouts on Edge Function, do not use it if you can avoid it.**
> 
> **to:**
> 
> ```ts
> import { sql } from "@vercel/postgres";
> 
> const result = await sql`SELECT now()`;
> // or const result = await sql.query('SELECT now()');
> ```
> 
> **👆This works and doesn't trigger timeouts.**
> 
> We understand that this is not ideal, especially because the drizzle-orm currently uses the problematic pattern. We understand not everyone can switch to this pattern.
> 
> > I tried with neondatabase directly (what vercel uses under the hood) and the same issue occurs However just like you I have no problem when switching to planetscale
> > Probably not a vercel issue in this case
> 
> Dear @chungweileong94 and @alexis-dorosz, if you were able to reproduce this issue directly on neon.tech, could you provide us with some code samples you used that reproduced the issue? Thanks.

hi, is there any workaround for prisma?

**Comment by vvo** [comment]
Hi @ghostneneji, can you give me pointers as for how you're using prisma + Vercel Postgres? Thanks

**Comment by vvo** [comment]
Hey there, as far as `@vercel/postgres` npm package goes, our latest release fixed this issue: https://github.com/vercel/storage/releases/tag/%40vercel%2Fpostgres%400.3.2
You only have to upgrade. If not just let us know in this thread 👍

**Comment by Xexr** [comment]
Thank you for the update, good to know it’s resolved.
